### ISSUE TITLE
S2187: False Positive with attributes derived from TestMethod (MSTest)

### ISSUE THREAD

**Issue by ThorstenReichert** [SonarSource/sonar-dotnet#4759]
### Description

Using the MSTest framework for unit testing, whever a `[TestClass]` contains only test methods using an attribute derived from `TestMethodAttribute`, the test class is flagged as containing no tests (S2187).

### Repro

```csharp
[TestClass]
public class UnitTest1
{
    [DerivedTestMethod]
    public void TestMethod1()
    {
        Assert.IsFalse(false);
    }
}

public class DerivedTestMethodAttribute : TestMethodAttribute
{
}
```

### Expected behavior

No warning should be raised.

### Actual behavior

S2187 (add some tests to this class) is raised on `UnitTest1`.

### Related information

* Visual Studio 16.10.3
* netcoreapp 3.1
* SonarAnalyzer.Csharp 8.26.0.34506
* Microsoft.Net.TestSdk 16.9.4 / MSTest.TestAdapter 2.2.3 / MSTest.TestFramework 2.2.2
* Windows 10
